Sources/Stevenson/GitHubService.swift

```diff
@@ -1,69 +1,108 @@
 import Foundation
 import Vapor
 
-public struct GitHubService {
-    private let baseURL = URL(string: "https://api.github.com:443")!
-    private let headers: HTTPHeaders
+public struct GitHubService: Service {
+    public let baseURL = URL(string: "https://api.github.com:443")!
+    public let headers: HTTPHeaders
 
     public init(username: String, token: String) {
-        let base64Auth = Data("\(username):\(token)".utf8).base64EncodedString(options: [])
-        self.headers = [
-            "Authorization": "Basic \(base64Auth)",
-            "Accept": "application/json"
-        ]
+        var headers = HTTPHeaders()
+        headers.add(name: HTTPHeaderName.accept, value: MediaType.json.description)
+        headers.basicAuthorization = BasicAuthorization(username: username, password: token)
+        self.headers = headers
     }
 }
 
 extension GitHubService {
-    public struct CommitList: Content {
-        let total_commits: Int
-        let commits: [Commit]
+    public struct Repository {
+        public let fullName: String
+        public let baseBranch: String
+        /// Release tags for this repository should match this regular expression 
+        /// and should capture version number in the first capture group.
+        public let releaseTag: String
 
-        func allMessages(includeDetails: Bool) -> [String] {
-            return self.commits.map {
-                let message = $0.commit.message
-                if includeDetails {
-                    return message
-                } else {
-                    return String(message[..<(message.firstIndex(of: "\n") ?? message.endIndex)])
-                }
-            }
+        public init(
+            fullName: String,
+            baseBranch: String,
+            releaseTag: String = "^([0-9]+.[0-9]+.[0-9]+)$"
```

Comment:
Note: We're actually discussing this format on Monday morning: for monorepos like us, tags will also contain the `appName`, so we might want to capture it in the optional 2nd capture group of the regex (actually in our case, the first) to populate `Release.appName`

Since we'll need to capture two things (the `Release.version` and `Release.appName`) and that depending on the convention used on each repo, we should not assume order of those captures (some might choose "1.2.3_myapp" while others might prefer "myapp/1.2.3" formats), maybe we should instead rely on named capture groups then?

Reply:
yea, it's not being used in this branch though, so can be removed for now